### PR TITLE
Remove jest-puppeteer in e2e

### PR DESCRIPTION
**Pull request by aweng98**
Code experiment:
- Most tests are deleted except few as examples.
- jest-puppeteer libs, setup and references are deleted.

### CODE REVIEW

File: e2e/.eslintrc.js [hunk 1]
```diff
@@ -16,10 +16,6 @@ module.exports = {
     browser: true
   },
   globals: {
-    page: true,
-    browser: true,
-    context: true,
-    jestPuppeteer: true
```
Comment (aweng98) [inline e2e/.eslintrc.js:22]: Getting rid of jest-puppeteer global objects references.

File: e2e/libs/page-init.ts [hunk 2]
```diff
@@ -139,21 +174,130 @@ beforeEach(async () => {
...
+      logger.error(err.message);
+    }
+    throw err;
+  } finally {
+    await browser.close().catch((err) => {
+      console.error(`Unable to close page. Error message: ${err.message}`);
+    });
+    await new Promise((resolve) => setTimeout(resolve, 1000));
+  }
+};
```
Comment (aweng98) [inline e2e/libs/page-init.ts:212]: `withBrowser` and `withPage` func are new.

File: e2e/tests/home/home-page.spec.ts [hunk 3]
```diff
@@ -1,50 +1,19 @@
...
-      const snowmanMenu = await card.getSnowmanMenu();
-      // Assumption: test user is workspace Owner.
-      // Check Workspace Actions snowman menu displayes the right set of options.
-      const links = await snowmanMenu.getAllOptionTexts();
-      expect(links).toEqual(expect.arrayContaining(['Share', 'Edit', 'Duplicate', 'Delete']));
-    }
+    await withPage()(async (page) => {
+      await signIn(page, config.readerUserName, config.userPassword);
+      await checkCreateNewWorkspaceLink(page);
+    });
```
Comment (aweng98) [inline e2e/tests/home/home-page.spec.ts:12]: Example of login by entering email and password.

File: e2e/tests/misc/sidenav.spec.ts [hunk 4]
```diff
@@ -1,51 +1,25 @@
...
-    const displayedUsername = await homePage.getUsername();
-    expect(displayedUsername).toBe(`${fname} ${lname}`);
-
-    // Select Your Workspaces link
-    await Navigation.navMenu(page, NavLink.YOUR_WORKSPACES);
-    const workspacesPage = new WorkspacesPage(page);
-    await workspacesPage.waitForLoad();
-    expect(await workspacesPage.isLoaded()).toBe(true);
+    await withPage()(async (page) => {
+      await signInWithAccessToken(page);
```
Comment (aweng98) [inline e2e/tests/misc/sidenav.spec.ts:10]: Example of login by access token.

File: e2e/tests/user/login.spec.ts [hunk 5]
```diff
@@ -1,26 +1,36 @@
...
+import { withPage } from 'libs/page-init';
 
 describe('Login tests:', () => {
   test('Cookie banner visible on login page', async () => {
-    const loginPage = new GoogleLoginPage(page);
-    await loginPage.load();
-    const link = await loginPage.cookiePolicyLink();
-    expect(link).toBeTruthy();
-    await link.click();
+    await withPage()(async (page, browser) => {
```
Comment (aweng98) [inline e2e/tests/user/login.spec.ts:8]: Example of accessing `page` and `browser` objects.

**Comment by aweng98** [comment]
Can you let me know what you think and how it can be better? Thank you.

File: e2e/libs/page-init.ts [hunk 6]
```diff
@@ -139,21 +174,80 @@ beforeEach(async () => {
...
+      console.log('withPage begin');
+      incognitoPage = await browser.createIncognitoBrowserContext().then((context) => context.newPage());
+      await setupPageBeforeEachTest(incognitoPage);
+      await new GoogleLoginPage(incognitoPage).load();
+      return await testFn(incognitoPage, browser);
+    } catch (err) {
+      if (err instanceof Error) {
+        logger.error(err.message);
+      }
+      await fs.ensureDir(failScreenshotDir);
```
Comment (calbach) [inline e2e/libs/page-init.ts:233]: All of this stuff does feel a bit unrelated to a function called `withPage()`. Something like `rwTest()` might capture the intent better, but I'm also OK with the current naming. Alternatively , is there no other hook available for generic failure handling in puppeteer test cases?

File: e2e/libs/page-init.ts [hunk 7]
```diff
@@ -139,21 +174,80 @@ beforeEach(async () => {
...
+      await fs.ensureDir(failHtmlDir);
+      const testName = testNames[testNames.length - 1];
```
Comment (calbach) [inline e2e/libs/page-init.ts:235]: This looks wrong, but might be WIP

File: e2e/libs/test-error-manager.ts [hunk 8]
```diff
@@ -0,0 +1,26 @@
...
+        reject(false);
+      } else {
+        console.info('Saved html file: ' + htmlFile);
+        resolve(true);
+      }
+    });
+  });
+};
+
+export const savePdf = async (page: Page, pdfFile: string) => {
```
Comment (calbach) [inline e2e/libs/test-error-manager.ts:24]: Unrelated:  I missed the addition of this - why are we saving PDFs now?

**Review by calbach — COMMENTED** [review]
I think the pattern looks reasonable - I assume the test code removal is temporary to get things compiling

**Pull request closed without merging**